iothub/device/src/ClientFactory.cs

```diff
@@ -88,7 +88,7 @@ public static InternalClient Create(string hostname, string gatewayHostname, IAu
                 throw new ArgumentException("Certificate chains are only supported on Amqp_Tcp_Only and Mqtt_Tcp_Only");
             }
 
-            IotHubConnectionStringBuilder connectionStringBuilder = IotHubConnectionStringBuilder.Create(hostname, gatewayHostname, authenticationMethod);
+            using IotHubConnectionStringBuilder connectionStringBuilder = IotHubConnectionStringBuilder.Create(hostname, gatewayHostname, authenticationMethod);
```

Comment:
I don't think disposing this object here is going to work:
- we have an instance of IotHubConnectionStringBuilder, which contains a disposable property "Certificate".
- we set internalClient.Certificate to point to the same instance of Certificate as in the IotHubConnectionStringBuilder.
- we return the InternalClient instance to the caller, but dispose the IotHubConnectionStringBuilder once this method exits (=> the Certificate is also disposed).
- Now we have an internal client, which contains a reference to a Certificate that is already disposed.

Reply:
Hmm interesting. I looked at who was using the builder but not the certificate. Who can dispose the builder then! The customer surely won't. Maybe needs to be disposed by the InternalClient and we hold a ref to it?

Reply:
Ok this was way more complicated than this. Worked offline with @abhipsaMisra and now we have something that seems correct.

Reply:
Will be testing this against the full pipeline before checking in.